HAL_ChibiOS: Added support for CORVON405V2.1

holydust opened the pull request:
new hardware added CORVON405V2.1
thx

holydust commented:
> commit title should be Tools: reserve bd id for CORVON405V2.1

sure, I have changed it, thx for ur reply

andyp1per commented:
please squash the commits into a single commit

peterbarker commented:
> please squash the commits into a single commit

Belay that.  This PR, despite the title, is adding the board, the id reservation and images.

So we do need three commits:
`board_types.txt: ...`
`hwdef: add CORVO...`

...also, please update the PR title, @holydust :-)

... we may also need the bootloader in here...

holydust commented:
> > please squash the commits into a single commit
> 
> Belay that. This PR, despite the title, is adding the board, the id reservation and images.
> 
> So we do need two commits: `board_types.txt: ...` `hwdef: add CORVO...`
> 
> ...also, please update the PR title, @holydust :-)
> 
> ... we may also need the bootloader in here...

Thx a lot Peter, I have changed the title, but I don't know how to squash the useless commits that I create

Hwurzburg commented:
@holydust I have fixed,squashed,and split...I will try to review content this week....

Hwurzburg reviewed (dismissed):
I corrected several errors:
1. no bootloader
2. BD ID already taken
3. periods not allowed in bd names...moved to CORVON405V2_1 for directory name and bd name
4. Moved most default params to hwdef to save flash
5. Updated readme with text needed for wiki when merged and made corrections
6. Changed OSD_TYPE to 5 to cover both internal and DisplayPort
pushed those changes and updated library splits

Should be good to go once @andyp1per approves

holydust commented:
> @holydust I have fixed,squashed,and split...I will try to review content this week....

thx a lot

holydust commented:
> I corrected several errors:
> 
> 1. no bootloader
> 2. BD ID already taken
> 3. periods not allowed in bd names...moved to CORVON405V2_1 for directory name and bd name
> 4. Moved most default params to hwdef to save flash
> 5. Updated readme with text needed for wiki when merged and made corrections
> 6. Changed OSD_TYPE to 5 to cover both internal and DisplayPort
>    pushed those changes and updated library splits
> 
> Should be good to go once @andyp1per approves

thx

andyp1per reviewed (changes requested):
It looks good but the AP_Bootloader and bootloaders changes need to be two separate commits with module titles "AP_Bootloader" and "bootloaders"

holydust commented:
> It looks good but the AP_Bootloader and bootloaders changes need to be two separate commits with module titles "AP_Bootloader" and "bootloaders"

sure, I have changed the title, hope this could go through well, thx for your time.

Hwurzburg commented:
@holydust that is not what Andy meant (and his requested change to separate the two commits to the Tools library is not a hard requirement at this time) was to organize the commits like this:
AP_Bootlaoder: add CORVON405V2_1   for the build_types.txt
bootloaders: add CORVON405V2_1 for the bootloader bin
AP_HAL_ChibiOS: add CORVON405V2_1 for all the other files

sorry for the churn...

holydust commented:
> @holydust that is not what Andy meant (and his requested change to separate the two commits to the Tools library is not a hard requirement at this time) was to organize the commits like this: AP_Bootlaoder: add CORVON405V2_1 for the build_types.txt bootloaders: add CORVON405V2_1 for the bootloader bin AP_HAL_ChibiOS: add CORVON405V2_1 for all the other files
> 
> sorry for the churn...

sorry for the misunderstanding, I've made them organized, hope it would work, thank u.

holydust commented:
@peterbarker Hello Peter, could you please take look at this PR, thx

Hwurzburg commented:
@holydust  by force pushing,there is no way to know what changes you just made,invalidating previous reviews...exactly what did you change?

holydust commented:
> @holydust by force pushing,there is no way to know what changes you just made,invalidating previous reviews...exactly what did you change?

@Hwurzburg sorry for the inconvenience, I didn't change anything, just clicked the "update branch" which I thought should keep it update :(

holydust commented:
> > @holydust by force pushing,there is no way to know what changes you just made,invalidating previous reviews...exactly what did you change?
> 
> @Hwurzburg sorry for the inconvenience, I didn't change anything, just clicked the "update branch" which I thought should keep it update :(

only rebased with the latest master

Hwurzburg commented:
no need to do that...it will be done on merge....if something ever causes a problem that requires an update tor ebase on master, it will be evident...

holydust commented:
> no need to do that...it will be done on merge....if something ever causes a problem that requires an update tor ebase on master, it will be evident...retagging for EU call

I got it,thx
